File: algorithms/102.Binary_Tree_Level_Order_Traversal/Binary_Tree_Level_Order_Traversa_dfs.py
# Definition for a  binary tree node
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

class Solution:
    # @param root, a tree node
    # @return a list of lists of integers
    def levelOrder(self, root):
        ret = []
        def dfs(root, level):
            if root:
                if level >= len(ret):
                    ret.append([])
                ret[level].append(root.val)
                dfs(root.left, level + 1)
                dfs(root.right, level + 1)

        dfs(root, 0)
        return ret

File: algorithms/102.Binary_Tree_Level_Order_Traversal/Binary_Tree_Level_Order_Traversa_dfs.cpp
#include <bits/stdc++.h>

using namespace std;

// Definition for binary tree
struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode(int x) : val(x), left(NULL), right(NULL) {}
};

class Solution {
private:
    vector<vector<int> >ans;
public:
    void DFS(TreeNode *curNode, int depth) {
        if (!curNode)
            return;
        // push back
        if (depth >= ans.size())
            ans.push_back(vector<int>(0));
        ans[depth].push_back(curNode->val);

        // recursive
        if (curNode->left)
            DFS(curNode->left, depth + 1);
        if (curNode->right)
            DFS(curNode->right, depth + 1);
    }

    vector<vector<int> > levelOrder(TreeNode *root) {
        ans.clear();
        DFS(root, 0);
        return ans;
    }
};

int main() {

    return 0;
}

